fpga_interchange/populate_chip_info.py

```diff
@@ -0,0 +1,500 @@
+                break
+
+            assert wire.name not in per_tile_map
+            per_tile_map[wire.name] = idx
+
+        tile_wire_to_wire_in_tile_index.append(per_tile_map)
+        num_tile_wires.append(max(per_tile_map.values())+1)
+
+    tiles = {}
+    tile_name_to_tile_index = {}
+
+    for tile_index, tile in enumerate(device.device_resource_capnp.tileList):
+        tile_info = TileInstInfo()
+
+        tile_info.name = device.strs[tile.name]
+        tile_info.type = tile.type
+        tile_info.tile_wire_to_node = list([-1 for _ in range(num_tile_wires[tile.type])])
+
+        tile_type = device.device_resource_capnp.tileTypeList[tile.type]
+
+        for site_type_in_tile_type, site in zip(tile_type.siteTypes, tile.sites):
+            site_name = device.strs[site.name]
+
+            # Emit primary type
+            site_info = SiteInstInfo()
+            site_type = device.device_resource_capnp.siteTypeList[site_type_in_tile_type.primaryType]
+            site_type_name = device.strs[site_type.name]
+            site_info.name = '{}.{}'.format(site_name, site_type_name)
+            site_info.site_type = site_type_name
+
+            tile_info.sites.append(len(chip_info.sites))
+            chip_info.sites.append(site_info)
+
+            for site_variant, (alt_site_type_index, _) in enumerate(zip(site_type.altSiteTypes, site_type_in_tile_type.altPinsToPrimaryPins)):
+                alt_site_info = SiteInstInfo()
+                alt_site_type = device.device_resource_capnp.siteTypeList[alt_site_type_index]
+                alt_site_type_name = device.strs[alt_site_type.name]
+                alt_site_info.name = '{}.{}'.format(site_name, alt_site_type_name)
+                alt_site_info.site_type = alt_site_type_name
+
+                tile_info.sites.append(len(chip_info.sites))
+                chip_info.sites.append(alt_site_info)
+
+        assert len(tile_info.sites) == chip_info.tile_types[tile.type].number_sites, (
+            tile_info.name,
+            len(tile_info.sites),
+            chip_info.tile_types[tile.type].number_sites
+            )
+
+        # (x, y) = (col, row)
+        tiles[(tile.col, tile.row)] = (tile_index, tile_info)
+
+    # Compute dimensions of grid
+    xs, ys = zip(*tiles.keys())
+    width = max(xs)+1
+    height = max(ys)+1
+
+    chip_info.width = width
+    chip_info.height = height
+
+    # Add tile instances to chip_info in row major order (per arch.h).
+    for y in range(height):
+        for x in range(width):
+            key = x, y
+
+            _, tile_info = tiles[key]
+            tile_name_to_tile_index[tile_info.name] = len(chip_info.tiles)
+            chip_info.tiles.append(tile_info)
+
+    # Output nodes
+    for idx, node in enumerate(device.device_resource_capnp.nodes):
+        # Skip nodes with only 1 wire!
+        if len(node.wires) == 1:
+            continue
+
+        node_info = NodeInfo()
+
+        # FIXME: Replace with actual node name?
```

Comment:
Do we need to handle the case where the node name isn't one of the wire names inside the node? If not; either we could say the node name always comes from wire index 0; or store an index to the 'primary' wire for naming purposes

Reply:
So I realized when debugging that RapidWright emits the node where the first wire in the node is the node name.  So I believe the bba as emitted will have the correct `getWireByName` that corresponds to the name of the node.

Reply:
> either we could say the node name always comes from wire index 0

To be clear, this is what the code is doing right now.